tooling-cpu-load/src/main/java/org/jboss/eap/qe/microprofile/tooling/cpu/load/utils/LinuxProcessUtils.java

```diff
@@ -0,0 +1,146 @@
+package org.jboss.eap.qe.microprofile.tooling.cpu.load.utils;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * Utility class for working with Linux process.
+ */
+public class LinuxProcessUtils {
+
+    /**
+     * Priority of scheduling process on CPU, can be set in range -20 to 20, lower number = higher priority
+     */
+    public static final int HIGHEST_CPU_PROCESS_PRIORITY = -20;
+    public static final int NORMAL_CPU_PROCESS_PRIORITY = 0;
+    public static final int LOWEST_CPU_PROCESS_PRIORITY = 20;
+
+    public static final String LINUX_OS_NAME = "Linux";
+
+    /**
+     * Retrieves process id of EAP/Wildfly server using managament client
+     *
+     * @param client management client connected to server
+     * @return pid of the server
+     */
+    public static int getProcessId(OnlineManagementClient client) throws IOException {
+        ModelNode model = new ModelNode();
+        model.get(ClientConstants.OP).set("read-resource");
+        model.get(ClientConstants.OP_ADDR).add("core-service", "platform-mbean");
+        model.get(ClientConstants.OP_ADDR).add("type", "runtime");
+
+        ModelNodeResult result = client.execute(model);
+        String nodeName = result.get("result").get("name").asString();
+        return Integer.valueOf(nodeName.substring(0, nodeName.indexOf("@")));
+    }
+
+    /**
+     * Returns pid of given process.
+     *
+     * @param process process
+     * @return pid of the process
+     */
+    public static int getProcessId(Process process) {
```

Comment:
Just a suggestions: Add a check if `process` is really instance of `java.lang.UNIXProcess` for faster fail and recognizable cause of fail.